### PR TITLE
Add an option for keeping trailing semicolons in rule

### PR DESCRIPTION
**Pull request by DChristiany**
See https://github.com/jakubpawlowicz/clean-css/issues/980.

I'm not certain I have implemented this in the way you wanted me too, so feel free to offer improvements

Thanks in advance

### CODE REVIEW

File: lib/options/format.js [hunk 1]
```diff
@@ -28,7 +28,8 @@ var DEFAULTS = {
   indentBy: 0,
   indentWith: IndentWith.Space,
   spaces: spaces(false),
-  wrapAt: false
+  wrapAt: false,
+  addSemicolonAfterLastPropertyInPropertyBlock: false
```
Comment (jakubpawlowicz) [inline lib/options/format.js]: I think we could name it `semicolonAfterLastProperty` as semicolons are not added anywhere else.

File: lib/writer/helpers.js [hunk 2]
```diff
@@ -79,7 +79,20 @@ function property(context, tokens, position, lastPropertyAt) {
...
+  if( context.format ) {
+    if( context.format.addSemicolonAfterLastPropertyInPropertyBlock || isPropertyBlock ) {
+      needsSemicolon = true;
+    } else if( position < lastPropertyAt ) {
+      needsSemicolon = true;
+    } else {
+      needsSemicolon = false;
+    }
+  } else {
+    needsSemicolon = position < lastPropertyAt || isPropertyBlock
```
Comment (jakubpawlowicz) [inline lib/writer/helpers.js]: Missing trailing semicolon (that's why Travis reported a failed build).

File: lib/writer/helpers.js [hunk 3]
```diff
@@ -79,7 +79,20 @@ function property(context, tokens, position, lastPropertyAt) {
   var store = context.store;
   var token = tokens[position];
   var isPropertyBlock = token[2][0] == Token.PROPERTY_BLOCK;
-  var needsSemicolon = position < lastPropertyAt || isPropertyBlock;
+
+  var needsSemicolon;
+  if( context.format ) {
```
Comment (jakubpawlowicz) [inline lib/writer/helpers.js]: Please use formatting consistent with all codebase, so `if (context.format) {`.

File: lib/writer/helpers.js [hunk 4]
```diff
@@ -79,7 +79,20 @@ function property(context, tokens, position, lastPropertyAt) {
...
+    if( context.format.addSemicolonAfterLastPropertyInPropertyBlock || isPropertyBlock ) {
```
Comment (jakubpawlowicz) [inline lib/writer/helpers.js]: ^ formatting

File: lib/writer/helpers.js [hunk 5]
```diff
@@ -79,7 +79,20 @@ function property(context, tokens, position, lastPropertyAt) {
...
+      needsSemicolon = true;
+    } else if( position < lastPropertyAt ) {
```
Comment (jakubpawlowicz) [inline lib/writer/helpers.js]: ^ formatting

**Comment by DChristiany** [comment]
Sorry, don't want to pressure you, I am just not sure, whether you were notified of my last commit

**Comment by jakubpawlowicz** [comment]
No worries, I think I didn't see the change. Looks good now! Merging 👍

**Comment by DChristiany** [comment]
Thanks a lot

**Comment by jakubpawlowicz** [comment]
Thanks for committing your time to do this. It'll end up in 4.2 release although you can safely use master if needed.

**Pull request merged**